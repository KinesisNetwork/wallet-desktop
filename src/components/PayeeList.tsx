import * as React from 'react'

import { LabelledField } from '@components/LabelledField'
import { Payee } from '@types'

type RemovePayee = (payeeName: string) => any
export interface Props {
  payees: Payee[]
  removePayee: RemovePayee
}

export const DeletePayeeLink: React.SFC<{ name: string, removePayee: RemovePayee }> = ({ name, removePayee }) => {
  return (
    <button className='button is-danger' onClick={() => removePayee(name)}>
      <span className='icon is-small'><i className='fas fa-trash-alt' /></span>
    </button>
  )
}

DeletePayeeLink.displayName = 'DeletePayeeLink'

export const PayeeList: React.SFC<Props> = (props) => {
  const payees = props.payees.map((payee) => {
    return (
      <LabelledField
        label={payee.name}
        key={payee.name}
        value={payee.publicKey}
        isClipped={true}
        addon={<DeletePayeeLink name={payee.name} removePayee={props.removePayee} />}
      />
    )
  })

  return (
    <div className='vertical-spaced'>
      <h1 className='title-heading has-text-centered'>MY PAYEES</h1>
      <section className='section'>
        {payees}
      </section>
    </div>
  )
}

PayeeList.displayName = 'PayeeList'
