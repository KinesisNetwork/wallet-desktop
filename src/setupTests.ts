import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
