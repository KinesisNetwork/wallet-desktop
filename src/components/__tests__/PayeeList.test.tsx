import { shallow } from 'enzyme'
import * as React from 'react'

import { DeletePayeeLink, PayeeList } from '@components/PayeeList'
import '../../setupTests'

describe('PayeeList', () => {
  let props

  beforeEach(() => {
    props = {
      payees: [
        {
          name: 'Fred',
          publicKey: '123',
        },
        {
          name: 'Mark',
          publicKey: '456',
        },
      ],
      removePayee: () => null,
    }
  })

  it('renders correctly', () => {
    const wrapper = shallow(<PayeeList {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should display a primary heading and two LabelledField components', () => {
    const wrapper = shallow(<PayeeList {...props} />)

    const heading = wrapper.find('h1')
    const labelledField = wrapper.find('LabelledField')

    expect(heading).toHaveLength(1)
    expect(heading.text()).toEqual('MY PAYEES')
    expect(labelledField).toHaveLength(2)
  })

  it('passes props down to LabelledField', () => {
    const wrapper = shallow(<PayeeList {...props} />)

    const labelledField = wrapper.find('LabelledField')

    expect(labelledField.at(0).prop('label')).toEqual('Fred')
    expect(labelledField.at(0).prop('value')).toEqual('123')
    expect(labelledField.at(1).prop('label')).toEqual('Mark')
    expect(labelledField.at(1).prop('value')).toEqual('456')
  })
})

describe('DeletePayeeLink', () => {
  let props

  beforeEach(() => {
    props = {
      name: 'Fred',
      removePayee: () => null,
    }
  })

  it('renders correctly', () => {
    const wrapper = shallow(<DeletePayeeLink {...props} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a button with an icon', () => {
    const wrapper = shallow(<DeletePayeeLink {...props} />)

    const button = wrapper.find('button')

    expect(button).toHaveLength(1)
    expect(button.children().find('.icon')).toHaveLength(1)
  })

  it('calls removePayee method on click', () => {
    const removePayeeMock = jest.fn()
    const wrapper = shallow(<DeletePayeeLink {...props} removePayee={removePayeeMock} />)

    const button = wrapper.find('button')
    button.simulate('click')

    expect(removePayeeMock).toHaveBeenCalledWith('Fred')
  })
})
