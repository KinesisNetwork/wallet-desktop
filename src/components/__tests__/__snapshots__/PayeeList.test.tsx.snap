// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeletePayeeLink renders correctly 1`] = `
<button
  className="button is-danger"
  onClick={[Function]}
>
  <span
    className="icon is-small"
  >
    <i
      className="fas fa-trash-alt"
    />
  </span>
</button>
`;

exports[`PayeeList renders correctly 1`] = `
<div
  className="vertical-spaced"
>
  <h1
    className="title-heading has-text-centered"
  >
    MY PAYEES
  </h1>
  <section
    className="section"
  >
    <LabelledField
      addon={
        <DeletePayeeLink
          name="Fred"
          removePayee={[Function]}
        />
      }
      isClipped={true}
      key="Fred"
      label="Fred"
      value="123"
    />
    <LabelledField
      addon={
        <DeletePayeeLink
          name="Mark"
          removePayee={[Function]}
        />
      }
      isClipped={true}
      key="Mark"
      label="Mark"
      value="456"
    />
  </section>
</div>
`;
